pymc3/backends/base.py

```diff
@@ -552,17 +553,25 @@ def merge_traces(mtraces):
 
     Raises
     ------
-    A ValueError is raised if any traces have overlapping chain numbers.
+    A ValueError is raised if any traces have overlapping chain numbers,
+    or if chains are of different lengths.
 
     Returns
     -------
     A MultiTrace instance with merged chains
     """
     base_mtrace = mtraces[0]
+    chain_len = len(base_mtrace)
```

Comment:
We should also explicitly raise an error if the list is empty.

Reply:
Adding check for `mtraces` being empty, and a simple test.